src/tools/listDatasources/listDatasources.ts

```diff
@@ -4,126 +4,135 @@ import { z } from 'zod';
 
 import { getConfig } from '../../config.js';
 import { getNewRestApiInstanceAsync } from '../../restApiInstance.js';
+import { Server } from '../../server.js';
```

Comment:
It looks like there are a lot of changes to each tool, but I've only introduced a factory method that returns the tool.